forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -122,6 +132,70 @@ fn validate_and_parse_salts<'a>(
     Ok(contract_salt_map)
 }
 
+async fn deploy_chunked(
+    command: &cmd::Deploy,
+    compiled: &BuiltPackage,
+    salt: Salt,
+    signing_key: &SecretKey,
+    provider: &Provider,
+    pkg_name: &str,
+) -> anyhow::Result<(ContractId, Vec<ContractId>)> {
+    println_action_green("Splitting", &format!("{pkg_name} into chunks"));
+    let contract_chunks = split_into_chunks(compiled.bytecode.bytes.clone(), MAX_CONTRACT_SIZE)?;
+    let mut deployed_contracts = vec![];
+    println_action_green("Deploying", &format!("{pkg_name} chunks"));
+    let chain_info = provider.chain_info().await?;
+    let target = Target::from_str(&chain_info.name).unwrap_or(Target::testnet());
+    let contract_url = match target.explorer_url() {
+        Some(explorer_url) => format!("{explorer_url}/contract/0x"),
+        None => "".to_string(),
+    };
+    for contract_chunk in contract_chunks {
+        let chunk_id = contract_chunk.id();
+        let deployed_chunk = contract_chunk.deploy(provider, signing_key).await?;
+        let chunk_contract_id = deployed_chunk.contract_id();
+        println_action_green(
+            "Finished",
+            &format!("deploying chunk {chunk_id} for {pkg_name}: {chunk_contract_id}"),
+        );
+        deployed_contracts.push(deployed_chunk);
+    }
+    println_action_green("Deployed", &format!("{pkg_name} chunks"));
+    let deployed_contract_ids: Vec<String> = deployed_contracts
+        .iter()
+        .map(|deployed_contract| format!("0x{}", deployed_contract.contract_id()))
+        .collect();
+
+    let deployed_contracts: Vec<_> = deployed_contracts
+        .iter()
+        .map(|deployed_contract| *deployed_contract.contract_id())
+        .collect();
+
+    let program_abi = match &compiled.program_abi {
+        sway_core::asm_generation::ProgramABI::Fuel(abi) => abi,
+        _ => bail!("contract chunking is only supported with fuelVM"),
+    };
+
+    println_action_green("Creating", &format!("loader contract for {pkg_name}"));
+    let loader_contract = build_loader_contract(
+        program_abi,
+        &deployed_contract_ids,
+        deployed_contracts.len(),
+        pkg_name,
+        &build_opts_from_cmd(command),
+    )?;
+
+    println_action_green("Deploying", &format!("loader contract for {pkg_name}"));
+    let deployed_id = deploy_pkg(command, &loader_contract, salt, provider, signing_key).await?;
+
+    println_action_green(
+        "Finished",
+        &format!("deploying loader contract for {pkg_name} {contract_url}{deployed_id}"),
+    );
+
+    Ok((deployed_id, deployed_contracts))
+}
+
```

Comment:
We need to update the logic that displays the number of transactions to include a transaction for each chunk.

![image](https://github.com/user-attachments/assets/6f9cd4e3-5ba4-4a0a-9b40-94ecf8d58874)